sys/net/network_layer/fib/fib.c

```diff
@@ -307,27 +330,39 @@ static int fib_remove(fib_entry_t *entry)
 /**
  * @brief signals (sends a message to) all registered routing protocols
  *        registered with a matching prefix (usually this should be only one).
- *        The message informs the recipient that no next-hop is available for the
- *        requested destination address.
  *        The receiver MUST copy the content, i.e. the address before reply.
  *
- * @param[in] dst       the destination address
- * @param[in] dst_size  the destination address size
+ * @param[in] type      the kind of signal
+ * @param[in] dat       the data to send
+ * @param[in] dat_size  the data size in bytes
+ * @param[in] dat_flags the data flags
  *
  * @return 0 on a new available entry,
  *         -ENOENT if no suiting entry is provided.
  */
-static int fib_signal_rp(uint8_t *dst, size_t dst_size, uint32_t dst_flags)
+static int fib_signal_rp(uint16_t type, uint8_t *dat, size_t dat_size, uint32_t dat_flags)
 {
     msg_t msg, reply;
-    rp_address_msg_t content;
-    content.address = dst;
-    content.address_size = dst_size;
-    content.address_flags = dst_flags;
+    rp_address_msg_t rp_addr_msg;
     int ret = -ENOENT;
+    void *content = NULL;
+
+    if (type != FIB_MSG_RP_SIGNAL_ROUTE_CREATED) {
```

Comment:
I find it quite hard to see the relationship between `FIB_MSG_RP_SIGNAL_ROUTE_CREATED` and source routes just by reading the code. Since `FIB_MSG_RP_SIGNAL_ROUTE_CREATED` is only used in the context of source routes, could you change the name to `FIB_MSG_RP_SIGNAL_SOURCE_ROUTE_CREATED`?